monai/bundle/config_item.py

```diff
@@ -19,13 +19,24 @@
 from collections.abc import Mapping, Sequence
 from importlib import import_module
 from pprint import pformat
-from typing import Any
+from typing import Any, Callable
 
 from monai.bundle.utils import EXPR_KEY
 from monai.utils import CompInitMode, ensure_tuple, first, instantiate, optional_import, run_debug, run_eval
 
 __all__ = ["ComponentLocator", "ConfigItem", "ConfigExpression", "ConfigComponent", "Instantiable"]
 
+from monai.utils.feature_flag import FeatureFlag
+
+CONFIG_COMPONENT_KEY_MODE = "_mode_"
```

Comment:
I would suggest using [StrEnum](https://github.com/Project-MONAI/MONAI/blob/daf2e714bb8a1cfa7c17d9aa7fb066dd37f5429e/monai/utils/enums.py#L69) for these variables so that they don't need `CONFIG_COMPONENT_` suffix and can be cast to `set` to produce `non_arg_keys` set below.